### PR TITLE
MULE-19596: Implement a profiling data consumer discovery strategy based on server plugins

### PR DESCRIPTION
**Pull request by IvanAndresFritzler**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategyTestCase.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mule.tck.size.SmallTest;
+
+@SmallTest
+public class CompositeProfilingDataConsumerDiscoveryStrategyTestCase {
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategyTestCase.java:26]: allure annotations,

File: core/src/main/java/org/mule/runtime/core/internal/config/builders/DefaultsConfigurationBuilder.java [hunk 2]
```diff
@@ -87,6 +87,8 @@
 import org.mule.runtime.core.internal.value.MuleValueProviderService;
 import org.mule.runtime.core.privileged.registry.RegistrationException;
 
+import java.util.Optional;
```
Comment (fsgonz) [inline core/src/main/java/org/mule/runtime/core/internal/config/builders/DefaultsConfigurationBuilder.java:90]: Not used.

File: core/src/main/java/org/mule/runtime/core/internal/profiling/DefaultProfilingService.java [hunk 3]
```diff
@@ -36,10 +42,11 @@
  */
 public class DefaultProfilingService extends AbstractProfilingService {
 
+  private Optional<Set<ProfilingDataConsumerDiscoveryStrategy>> profilingDataConsumerDiscoveryStrategies = Optional.empty();
```
Comment (fsgonz) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/DefaultProfilingService.java:45]: static import

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: core-tests/src/test/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategyTestCase.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        {
+          add(discoveryStrategyOne);
+          add(discoveryStrategyTwo);
+        }
+      };
+
+  @Before
+  public void setUp() {
+    when(discoveryStrategyOne.discover()).thenAnswer(invocation -> Collections.singleton(profilingDataConsumerOne));
+    when(discoveryStrategyTwo.discover()).thenAnswer(invocation -> Collections.singleton(profilingDataConsumerTwo));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategyTestCase.java:44]: static imports

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by fsgonz — APPROVED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategy.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * instances into a single one.
+ * 
+ * @since 4.5.0
+ */
+public class CompositeProfilingDataConsumerDiscoveryStrategy implements ProfilingDataConsumerDiscoveryStrategy {
+
+  /**
+   * Discovery strategies whose sets of discovered {@link ProfilingDataConsumer} will be returned by {@link #discover()}.
+   */
+  Set<ProfilingDataConsumerDiscoveryStrategy> discoveryStrategies;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/discovery/CompositeProfilingDataConsumerDiscoveryStrategy.java:26]: private?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**